base/sparse/sparsematrix.jl

```diff
@@ -9,10 +9,17 @@ type SparseMatrixCSC{Tv,Ti<:Integer} <: AbstractSparseMatrix{Tv,Ti}
     colptr::Vector{Ti}      # Column i is in colptr[i]:(colptr[i+1]-1)
     rowval::Vector{Ti}      # Row values of nonzeros
     nzval::Vector{Tv}       # Nonzero values
-end
 
-SparseMatrixCSC{Tv,Ti}(m::Integer, n::Integer, colptr::Vector{Ti}, rowval::Vector{Ti}, nzval::Vector{Tv}) =
-    SparseMatrixCSC(int(m), int(n), colptr, rowval, nzval)
+    function SparseMatrixCSC(m::Integer, n::Integer, colptr::Vector{Ti}, rowval::Vector{Ti}, nzval::Vector{Tv})
+        ((m < 0) || (n < 0)) && error("invalid Array dimensions")
```

Comment:
`ArgumentError` seems more appropriate. Do we document all the exceptions in the manual? That would help someone pick an appropriate one to throw.